strawberry/types/type_resolver.py

```diff
@@ -49,24 +49,147 @@ def _resolve_generic_type(type: Type, field_name: str) -> Type:
     return type
 
 
-def resolve_type(field_definition: Union[FieldDefinition, ArgumentDefinition]) -> None:
-    # convert a python type to include a strawberry definition, so for example
+def resolve_type_field(field: StrawberryField) -> None:
+    # TODO: This should be handled by StrawberryType in the future
+    if isinstance(field.type, str):
+        module = sys.modules[field.origin.__module__]
+
+        field.type = eval(field.type, module.__dict__)
+
+    if isinstance(field.type, LazyType):
+        field.type = field.type.resolve_type()
+
+    if is_forward_ref(field.type):
+        # if the type is a forward reference we try to resolve the type by
+        # finding it in the global namespace of the module where the field
+        # was initially declared. This will break when the type is not declared
+        # in the main scope, but we don't want to support that use case
+        # see https://mail.python.org/archives/list/typing-sig@python.org/thread/SNKJB2U5S74TWGDWVD6FMXOP63WVIGDR/  # noqa: E501
+
+        type_name = field.type.__forward_arg__
+
+        module = sys.modules[field.origin.__module__]
+
+        # TODO: we should probably raise an error if we can't find the type
```

Comment:
This TODO seems like an important one. Are you going to do it as part of this PR?

Reply:
This isn't really new code -- it's not related to this PR. 

Previously this function handled both `FieldDefinitions` and `ArgumentDefinitions` and I needed a way to treat the fields differently for this refactoring. I just cloned the function and started modifying the one that deals with fields. In the future, once `StrawberryType` is implemented both functions will disappear/be absorbed by `StrawberryType`.